### PR TITLE
ENH: polish scripts (error handling, paths)

### PR DESCRIPTION
**Pull request by ZLLentz**
## Description

This PR doesn't change the workflows at all, but makes small changes to the bash scripts for maintenance and consistency.

Changes:
- Remove the incorrect/incomplete python environment checks, replacing them with always sourcing the correct env (with an environment variable override option).
- Define the correct python env in a central file (`activate_python.sh`) instead of in each script that needs it.
- Centralize the path definitions in a central file (`paths.sh`) instead of in each script that needs them.
- Also in `paths.sh`, cd to the ansible root because this is the only directory where the ansible scripts work properly. This will let us effectively run the scripts from anywhere, which was previously intended but not possible.
- Make all the scripts exit on failure and do the ssh agent cleanup regardless of when they fail, using `set -e` and `trap`

## Motivation and Context

- I noticed that the python env sourcing wasn't quite correct
- I noticed that the ssh agent would sometimes not be cleaned up if a script exited early
- I noticed that some of the scripts would continue onward despite an issue
- I noticed that the scripts could only be run from the ansible root

And I wanted to fix all of these.

## How Has This Been Tested?

Interactively from a fresh BSD install and from already set up test PLCs.

## Where Has This Been Documented?

I'm not intending to add anything to the docs, this is business as usual but with edge case bugfixing.

## Pre-merge checklist
- [x] Code works interactively
- [x] Code works interactively in dry_run mode
- [x] Code contains descriptive docstrings, including context and API
- [x] Pre-commit passes on GitHub Actions
- [x] Documentation under https://confluence.slac.stanford.edu/display/PCDS/TcBSD has been updated appropriately

### CODE REVIEW

**Review by ghalym — APPROVED** [review]
lgtm!

File: scripts/activate_python.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+# Helper script for activating the correct python environment.
+# Sets a default env or you can provide your own activate path for testing
+
+DEFAULT_ENV=/cds/group/pcds/pyps/conda/venvs/ansible/bin/activate
+
+if [ -f "${ANSIBLE_PYTHON_ACTIVATE:=${DEFAULT_ENV}}" ]; then
+  source "${ANSIBLE_PYTHON_ACTIVATE}"
+else
+  echo "No Python activation script found at ${ANSIBLE_PYTHON_ACTIVATE}"
+  return 1
```
Comment (slactjohnson) [inline scripts/activate_python.sh:12]: Nitpick: Shouldn't we also return 0 if ANSIBLE_PYTHON_ACTIVATE was found?

**Review by slactjohnson — APPROVED** [review]
LGTM

File: scripts/activate_python.sh (hunk 1 above)
Comment (ZLLentz) [reply scripts/activate_python.sh:12]: return 0 is implied, when the script rolls off the end without erroring or returning early
